site/documentation/content/api/device-connection/index.md

```diff
@@ -107,6 +107,176 @@ Implementors of this API may return a *404* status code in order to indicate tha
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Get command-handling protocol adapter instances for device
+
+Clients use this command to *get* information about the adapter instances that can handle command & control messages for the given device, possibly using one of the given gateways.
+
+As part of this request, the client is supposed to provide the list of gateways that may act on behalf of the given device. This API doesn't mandate checks on the validity of the given device and the gateway list in order not to introduce a dependency on the *Device Registration API*.
+
+The command implementation MUST determine the adapter instances by applying the following rules (in the given order):
+- If an adapter instance is associated with the given device, this adapter instance is returned as the *single* returned list entry.
+- Otherwise, if there is an adapter instance registered for the last known gateway associated with the given device, this adapter instance is returned as the *single* returned list entry. The last known gateway has to be contained in the given list of gateways for this case.
+- Otherwise, *all* adapter instances associated with any of the given gateway identifiers are returned.
+
+That means that for a device communicating via a gateway, the result is reduced to a *single element* list if an adapter instance for the device itself or its last known gateway is found. The adapter instance registered for the device itself is given precedence in order to ensure that a gateway having subscribed to commands for that *particular device* is chosen over a gateway that has subscribed to commands for *all* devices of a tenant.
+
+**Message Flow**
+
+
+{{< figure src="get_cmd_handling_adapter_instances.svg" title="Client retrieving the list of command-handling adapter instances for a device" alt="A client sends a request message for retrieving the list of command-handling adapter instances for a device and receives a response containing the information" >}}
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to retrieve the command-handling adapter instances for a device in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name           | Mandatory | Location                 | AMQP Type | Description |
+| :------------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*      | yes       | *properties*             | *string*  | MUST be set to `get-cmd-handling-adapter-instances`. |
+| *gateway_ids*  | yes       | *properties*             | *string*  | A JSON array containing the IDs of the gateways that may act on behalf of the given device. |
```

Comment:
FMPOV we should put a JSON array containing the ids into the body of the request message ...
Otherwise, the location needs to be *application-properties* ...

Reply:
> Otherwise, the location needs to be *application-properties* ...

Right, that was a copy/paste error here.

Regarding using the request message instead: I had this in my initial commit here, but I think putting the operation parameters in the (application-) properties of the request message is more commonly used in our APIs.